### PR TITLE
Add tips & tricks documentation

### PR DESCRIPTION
**Pull request by jnikula**
Add some tips for using Sphinx and Hawkmoth for C/C++ API documentation, etc.

### CODE REVIEW

File: doc/tips.rst [hunk 1]
```diff
@@ -59,3 +59,21 @@ The Sphinx alternative is :external+sphinx:rst:dir:`literalinclude`. The
 .. _@include: https://www.doxygen.nl/manual/commands.html#cmdinclude
 
 .. _@snippet: https://www.doxygen.nl/manual/commands.html#cmdsnippet
+
+Dealing With Generated Headers
+------------------------------
```
Comment (BrunoMSantos) [inline doc/tips.rst:64]: This one is a bit much in my opinion, but I guess it's fine? I do see this as a workaround for a problem that is entirely unrelated to hawkmoth. If it were me though, I'd make the documentation depend on generating the headers, let the build system do what it's meant to do. :man_shrugging:

**Review by BrunoMSantos — APPROVED** [review]
Cool idea and nice set of tips. Apart from the one I objected to, but I'm ok with it really. I'll leave it up to you ;)

File: doc/tips.rst (hunk 1 above)
Comment (jnikula) [reply doc/tips.rst:64]: FWIW this is the approach taken in Mesa. I think it's a tall order to make the documentation build depend on generating the headers, which in turn depend on additional tooling and whatnot.

The reason this is related to Hawkmoth is that we use the compiler to extract the documentation, and therefore hit the the problem of missing headers. AFAIK Doxygen uses a parser of its own, and doesn't care if it can't find headers.

Comment (jnikula) [reply doc/tips.rst:64]: Arguably it's a problem in Mesa that the Sphinx build is not part of the Meson build files, at all. The documentation is generated simply by running `sphinx-build` in CI. But that's actually the same as on Read the Docs.

Comment (BrunoMSantos) [reply doc/tips.rst:64]: So, I'm still ok with this. But maybe it wouldn't trigger me as much if it were a more nuanced tip... One that points out that the files must be created somehow and that it should typically be handled by the build system and possibly with `conf.py`. And also that the code doesn't have to be fully compilable as we skip the function bodies, so the generated files can be stubs instead.

As it stands, I think this particular advise is a bit out of place with Hawkmoth's ethos if not bad general advise altogether. Using a compiler underneath is _the_ defining feature of hawkmoth, it stands to reason the code must be compilable.

If I were to advise on this topic, I would rather recommend 'use a build system, take the flags from there and ...':
- If performance is not an issue, generate the files.
- If performance is an issue, generate the smallest adequate stub from the same sources / tools (single source of truth and whatnot).

Also, if dependent on calling `sphinx-build` without build system, I'd still call the build system from the `conf.py` to generate those files.

Comment (jnikula) [reply doc/tips.rst:64]: Fair enough. I think I'll leave the section out altogether, at least for now.

**Comment by jnikula** [comment]
And merged. Thanks for your input @BrunoMSantos !
